scripts/base/protocols/ldap/main.zeek

```diff
@@ -0,0 +1,513 @@
+# Copyright (c) 2021 by the Zeek Project. See LICENSE for details.
+
+module LDAP;
+
+export {
+  redef enum Log::ID += { LDAP_LOG,
+                          LDAP_SEARCH_LOG };
+
+  ## TCP ports which should be considered for analysis.
+  const ports_tcp = { 389/tcp, 3268/tcp } &redef;
+
+  ## UDP ports which should be considered for analysis.
+  const ports_udp = { 389/udp } &redef;
+
+  redef likely_server_ports += { LDAP::ports_tcp, LDAP::ports_udp };
+
+  ## Whether clear text passwords are captured or not.
+  option default_capture_password = F;
+
+  ## Whether to log LDAP search attributes or not.
+  option default_log_search_attributes = F;
+
+  ## Default logging policy hook for LDAP_LOG.
+  global log_policy: Log::PolicyHook;
+
+  ## Default logging policy hook for LDAP_SEARCH_LOG.
+  global log_policy_search: Log::PolicyHook;
+
+  #############################################################################
+  # This is the format of ldap.log (ldap operations minus search-related)
+  # Each line represents a unique connection+message_id (requests/responses)
+  type Message: record {
+
+    # Timestamp for when the event happened.
+    ts: time &log;
+
+    # Unique ID for the connection.
+    uid: string &log;
+
+    # The connection's 4-tuple of endpoint addresses/ports.
+    id: conn_id &log;
+
+    # transport protocol
+    proto: string &log &optional;
+
+    # Message ID
+    message_id: int &log &optional;
+
+    # LDAP version
+    version: int &log &optional;
+
+    # normalized operations (e.g., bind_request and bind_response to "bind")
+    opcode: set[string] &log &optional;
+
+    # Result code(s)
+    result: set[string] &log &optional;
```

Comment:
Given `result`, `diagnostic_message`, `object`, and `argument` are all containers, I want to suggest pluralizing them to match other logs (conn: tunnel_parents, HTTP: tags, DNS: answers, TTLs, DHCP: msg_types).

Same for `scope`, `deref`, `base_object`, etc. in `Search`.

Can consider as follow-up with feedback from others.